drivers/entropy/entropy_mcux_trng.c

```diff
@@ -11,6 +11,14 @@
 
 #include "fsl_trng.h"
 
+/* The True Random Number Generator device is called "TRNG0" on the
+ * Kinetis SoCs and "TRNG" on the I.MX RT SoC.  Add a define to cope
+ * with that.
+ */
+#ifndef TRNG0
+#define TRNG0 TRNG
+#endif
+
```

Comment:
I don't think it's a good idea. After checking the code, I would suggest not to hard coding TRNG0 in this file, but to use something like  TRNG_BASE_PTRS[0] as a workaround, given the different macro of this base address.

Reply:
Thanks for your suggestion, I have just pushed a new version fixing that.

Reply:
Actually you should get the base pointer from device tree instead. Please make a separate commit to change the driver to use `DT_ENTROPY_MCUX_TRNG_BASE_ADDRESS` below, then a second commit to enable the driver on rt.

Reply:
I didn't realize that `TRNG` and `TRNG0` corresponded to the base address of the peripheral, thanks for the hint. I have just reworked the PR as you requested.